tests/e2e/workloads/amq/test_rgw_kafka_notifications.py

```diff
@@ -0,0 +1,185 @@
+
+
+def clone_notify():
+    """
+    Returns notify path
+    """
+    notify_dir = tempfile.mkdtemp(prefix="notify_")
+    log.info(f"cloning repo notify in {notify_dir}")
+    git_clone_cmd = f"git clone {constants.RGW_KAFKA_NOTIFY}"
+    run(git_clone_cmd, shell=True, cwd=notify_dir, check=True)
+    notify_path = f"{notify_dir}/notify/notify.py"
+    return notify_path
+
+
+@tier1
+@bugzilla("1937187")
+@vsphere_platform_required
+@pytest.mark.parametrize(
+    argnames="interface",
+    argvalues=[
+        pytest.param(*["RGW-OC"], marks=[pytest.mark.polarion_id("OCS-2514")]),
+    ],
+)
+class TestRGWAndKafkaNotifications(E2ETest):
+    """
+    Test to verify rgw kafka notifications
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def test_fixture_amq(self, request):
+        self.amq = AMQ()
+
+        self.kafka_topic = (
+            self.kafkadrop_pod
+        ) = self.kafkadrop_svc = self.kafkadrop_route = None
+
+        def teardown():
+
+            if self.kafka_topic:
+                self.kafka_topic.delete()
+            if self.kafkadrop_pod:
+                self.kafkadrop_pod.delete()
+            if self.kafkadrop_svc:
+                self.kafkadrop_svc.delete()
+            if self.kafkadrop_route:
+                self.kafkadrop_route.delete()
+
+            self.amq.cleanup()
+
+        request.addfinalizer(teardown)
+        return self.amq
+
+    def test_rgw_kafka_notifications(self, interface, bucket_factory):
+        """
+        Test to verify rgw kafka notifications
+
+        """
+        # Get sc
+        sc = default_storage_class(interface_type=constants.CEPHBLOCKPOOL)
+
+        # Deploy amq cluster
+        self.amq.setup_amq_cluster(sc.name)
+
+        # Create topic
+        self.kafka_topic = self.amq.create_kafka_topic()
+
+        # Create kafkadrop pod
+        try:
+            kafkadrop = list(
+                templating.load_yaml(constants.KAFKADROP_YAML, multi_document=True)
+            )
+            self.kafkadrop_pod = OCS(**kafkadrop[0])
+            self.kafkadrop_svc = OCS(**kafkadrop[1])
+            self.kafkadrop_route = OCS(**kafkadrop[2])
+            self.kafkadrop_pod.create()
+            self.kafkadrop_svc.create()
+            self.kafkadrop_route.create()
+        except (CommandFailed, CalledProcessError) as cf:
+            log.error("Failed during creation of kafkadrop which kafka UI")
+            raise cf
+
+        # Validate kafkadrop pod running
+        ocp_obj = OCP(kind=constants.POD, namespace=constants.AMQ_NAMESPACE)
+        ocp_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector="app=kafdrop",
+            timeout=120,
+            sleep=5,
+        )
+
+        # Get the kafkadrop route
+        kafkadrop_host = self.kafkadrop_route.get().get("spec").get("host")
+
+        # # Get the kafkaendpoint
+        kafka_endpoint = f"my-cluster-kafka-bootstrap.{constants.AMQ_NAMESPACE}.svc.cluster.local:9092"
```

Comment:
Can kafka_endpoint variable moved under constants?

Reply:
moved to constants